src/ducks/classifications.js

```diff
@@ -121,10 +118,128 @@ const createClassification = () => {
+      - If the item (Classification) is successfully submitted/saved, that's it.
+        (It's effectively removed from the queue.)
+      - If the item fails, it's reinserted into the new queue, to be processed
+        the next time a new Classification is submitted.
+    - Once ALL the items are processed, we are now ready to prepare our data
+      for the next Subject to be fetched.
+ */
+const saveAllQueuedClassifications = (dispatch, user = null) => {
+  const QUEUE_NAME = (user)
+    ? user.id + '.' + CLASSIFICATIONS_QUEUE_NAME
+    : '_.' + CLASSIFICATIONS_QUEUE_NAME;
+        
+  const queue = JSON.parse(localStorage.getItem(QUEUE_NAME));
+
+  if (queue && queue.length !== 0) {
+    //Prepare 
+    const newQueue = [];
+    localStorage.setItem(QUEUE_NAME, null);  //Empty the queue first
+    
+    //Keep track of items processed so we know when ALL items are processed.
+    let itemsProcessed = 0;
+    let itemsFailed = 0;
+    const itemsToProcess = queue.length;
+    
+    queue.forEach((classificationData) => {
+      //Let's send those Classifications!
+      apiClient.type('classifications').create(classificationData).save()
+      
+      //SUCCESS
+      .then((classificationObject) => {
+        console.info('ducks/classifications.js saveAllQueuedClassifications() success: item ', classificationObject.id);
+
+        try {
+          Split.classificationCreated(classificationObject);
+        } catch (err) { console.error('Split.classificationCreated() error: ', err); }
+
+        //Record locally that the Classification has been seen.
+        const { workflow, subjects } = classificationObject.links;
+        dispatch(addAlreadySeen(workflow, subjects));
+        
+        //OK, we don't need you any more, Classification. Why is this here? It's in PFE.
+        //TODO: check
+        classificationObject.destroy();
+      })
+      
+      //FAILURE
+      .catch((err) => {
+        //Ah, crap.
+        itemsFailed++;
+        
+        console.error('ducks/classifications.js saveAllQueuedClassifications() failure: item ', classificationObject.id);
+        console.error('ducks/classifications.js saveAllQueuedClassifications() error: ', err);
+        Rollbar && Rollbar.error &&
+        Rollbar.error('ducks/classifications.js saveAllQueuedClassifications() error: ', err);
+
+        //Right, why did it fail?
+        switch (err.status) {
+          case 422:  //If the reason for failure is that Panoptes returned a 422, it means the Classification was bad and should be discarded.
+            break;
+
+          default:  //Otherwise, the failed Classification should be re-queued for the next time attempt.
+            newQueue.push(classificationData);
+        }
+      })
+      .finally(() => {
+        //That's one item down.
+        itemsProcessed++;
+
+        //Have all items been processed?
+        if (itemsProcessed === itemsToProcess) {
+          //Did anything fail?
+          if (itemsFailed > 0) {
```

Comment:
What if two queued items fail? Will they get this alert twice?

Reply:
Oh wait, disregard this. I just noticed the indent when looking at this on Atom.

Reply:
No, because the alert code will only be called when _all items_ have finished processing.

```
//Check that we're done with everything first
if (itemsProcessed === itemsToProcess) {
  //Did anything fail?
  if (itemsFailed > 0) { ...
```